### PR TITLE
Add go mod and update imports

### PR DESCRIPTION
**Pull request by fierlion**
### Summary
This adds go mod integration and updates the imports.  Specifically the `aws-sdk-go` is updated to the latest available version at the time of this change.   The `containernetworking` imports are meant to be minimally invasive to the existing code.

### Implementation details
- The specific container networking imports are based on the versions used in the existing code. The base `containernetworking/cni` package was broken into `/cni` and `/plugins` and the imports have been updated accordingly.
- The mocks have been regenerated.
- To integrate with `go mod` the following steps were completed:
```
go mod init
#manual updates made to the versions for cni and aws-sdk-go
go mod tidy #download the changes
go mod vendor #update the vendor dir
```
- Where necessary, individual changes were made to various functions to adhere to the existing code logic.
These include:
-- pkg/cniipwrapper/ip.go -> removing input `family int` and updated return logic the `DelLinkByNameAddr` as the function signature has changed.
-- plugins/ecs-bridge/commands/commands.go -> update the configure interface command which requires an *int now.
-- updated all plugin main functions -> `skel.PluginMain(commands.Add, commands.Check, commands.Del, cnispec.GetSpecVersionSupported(), bv.BuildString("TODO"))` <- the PluginMain interface changed to require the `Check` command and the build version string.  For now the Check function is a placeholder.

- Finally `go fmt ./...` was run on all.

### Testing
make unit-test
make integration-test

New tests cover the changes: no

### Description for the changelog
Add go mod and update imports.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/ipam/commands/commands_test.go [hunk 1]
```diff
@@ -70,7 +71,6 @@ func TestGetSpecificIPV4HappyPath(t *testing.T) {
 
 	gomock.InOrder(
 		allocator.EXPECT().Assign(gomock.Any(), gomock.Any()).Return(nil),
-		allocator.EXPECT().SetLastKnownIP(net.ParseIP("10.0.0.3")),
```
Comment (fierlion) [inline plugins/ipam/commands/commands_test.go:73]: https://github.com/aws/amazon-ecs-cni-plugins/blob/55b2ae77ee0bf22321b14f2d4ebbcc04f77322e1/plugins/ipam/commands/commands.go#L160-L168

removed this as the `else` statement is not touched.  It has a call to `getIPV4AddressFromDB` which calls `SetLastKnownIP`

File: pkg/cniipwrapper/ip.go [hunk 2]
```diff
@@ -47,6 +47,11 @@ func (*cniIP) SetHWAddrByIP(ifName string, ip4 net.IP, ip6 net.IP) error {
 	return ip.SetHWAddrByIP(ifName, ip4, ip6)
 }
 
-func (*cniIP) DelLinkByNameAddr(ifName string, family int) (*net.IPNet, error) {
-	return ip.DelLinkByNameAddr(ifName, family)
+func (*cniIP) DelLinkByNameAddr(ifName string) (*net.IPNet, error) {
```
Comment (yinyic) [inline pkg/cniipwrapper/ip.go:50]: It doesn't seem like we make use of the IP address return (in `deletelink.go`). Maybe we can just return the error.

File: pkg/cniipwrapper/ip.go [hunk 3]
```diff
@@ -16,20 +16,20 @@ package cniipwrapper
...
 
-	"github.com/containernetworking/cni/pkg/ip"
-	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/containernetworking/plugins/pkg/ip"
+	"github.com/containernetworking/plugins/pkg/ns"
 )
 
 // IPAM wraps methods used from the the cni/pkg/ip package
-// Seee github.com/containernetworking/cni/pkg/ip for more details
+// Seee github.com/containernetworking/plugins/pkg/ip for more details
```
Comment (yinyic) [inline pkg/cniipwrapper/ip.go:24]: Nit: `Seee` -> `See`

File: plugins/ecs-bridge/engine/engine.go [hunk 4]
```diff
@@ -190,8 +190,7 @@ func (engine *engine) RunIPAMPluginAdd(plugin string, netConf []byte) (*current.
 
 	result, err := current.NewResultFromResult(ipamResult)
 	if err != nil {
-		return nil, errors.Wrapf(err,
-			"bridge IPAM ADD: unable to parse result '%s'", ipamResult.String())
+		return nil, errors.Wrapf(err, "bridge IPAM ADD: unable to parse result.")
```
Comment (yinyic) [inline plugins/ecs-bridge/engine/engine.go:193]: Why are we making the change? I'd imagine that printing `ipamResult` could be helpful in understanding why we failed to parse?

Comment (fierlion) [reply plugins/ecs-bridge/engine/engine.go]: updated and added test.

File: pkg/cniipwrapper/ip.go (hunk 3 above)
Comment (fierlion) [reply pkg/cniipwrapper/ip.go]: updated

**Review by YashdalfTheGray — APPROVED** [review]
Walked through this PR on slack, reviewed it offline and focused on the implementation details section. Scrolled through the rest of the files besides the mocks (mostly import path changes) and Ray also ran integ tests.

**Review by yinyic — APPROVED** [review]

**Pull request merged**